lab.R

```diff
@@ -101,25 +106,49 @@ daily_diff = function(x, order_var) {
 #' *Write a pipe that takes `filter_df` as input, groups the data by county and FIPS code, sorts the dataframe by date, and then converts the cumulative cases and death counts to daily changes using `daily_diff`.  (`date` is the order variable.)  Assign the result to `daily_df`. Hint: `mutate()` can replace the value of existing variables.* 
 #' 
 
+daily_df = filtered_df %>% 
+  group_by(county, fips) %>% 
+  mutate(cases = daily_diff(cases, date), deaths = daily_diff(deaths, date)) %>% 
+  ungroup()
+
+#when I was viewing this before ungroup the rows were still not grouped by those variables. Am i using group by wrong or misunderstanding how it works?
+
 #' 3. *Finally we need to calculate rates per 1 million residents.  Write a pipe that takes `daily_diff` as input, joins it with the `pop` dataframe using appropriate variables, removes any rows with missing FIPS codes, and constructs the variables `cases_per_pop` and `deaths_per_pop`.  When constructing these variables, multiply by `per_pop` to get rates per 1 million residents.  Assign the result to `covid_df`, since this contains the Covid data for our analysis.*  
 #' 
 
+covid_df = daily_df %>% 
+  left_join(pop, by = c('state', 'county', 'fips')) %>% 
+  filter(!is.na(fips)) %>% 
+  mutate(cases_per_pop = cases/population * per_pop, deaths_per_pop = deaths/population * per_pop)
+  
+# there are many na values fro cases and deaths in this. Did I lose some or did I misunderstand the nominally complete part before?
 
 
 #' # Problem 4 #
 #' 1. *To explore these time-series data visually, we'll want to use line plots of cases or deaths over time.  The line group needs the `group` aesthetic to determine which values should be treated as part of a single line.  Uncomment and fill in the blanks to plot cases per 1,000,000 residents over time for each county.*  
 #' 
 
-# ggplot(covid_df, aes(---, ---, group = ---)) +
-#     geom_line()
+ggplot(covid_df, aes(date, cases_per_pop, group = county)) +
+    geom_line() +
```

Comment:
With this hanging `+`, R tries to combine this ggplot object with the thing next (the ggplot object on line 137).  It can't do this so the lab script fails before running the tests.  This is why `testthat::test_dir('tests')` is failing. 
```suggestion
    geom_line()
```